app/controllers/lti_assignment_users_controller.rb

```diff
@@ -0,0 +1,221 @@
+      if(user)
+        assignment_id = params['custom_assignment_id'] # assignment_id needs to be passed as a
+        assignment = Assignment.find(assignment_id)
+        if assignment
+          # Assignment number passed in the request is valid.
+          assignment_participant = AssignmentParticipant.find_by_user_id_and_assignment_id(user.id,assignment_id)
+          lis_result_sourcedid = params["lis_result_sourcedid"]; # sourcedid is used to send grades back
+          @tenant.lis_outcome_service_url = params['lis_outcome_service_url'];
+          unless assignment_participant
+            # Assignment_Participant not found in participant table and assignment exists. So add user to participant table.
+            assignment.add_participant_by_user(user, true, true, true) # By default, set can_review, can_quiz, can_submit as true
+            assignment_participant = AssignmentParticipant.find_by_user_id_and_assignment_id(user.id, assignment_id)
+          end
+          if lis_result_sourcedid and @tenant.lis_outcome_service_url
+            # Add participant, assignment to LtiAssignmentUser table for enabling "grade passback" feature
+            # LtiAssignmentUser is a mapping of the sourcedid's with Expertiza assignment and user
+            ua = LtiAssignmentUser.find_by_participant_id assignment_participant # LtiAssignmentUser already exist for participant, so just update.
+            if ua == nil
+              ua = LtiAssignmentUser.new
+            end
+            ua.assignment_id = assignment_id;
+            ua.user_id = user.id;
+            ua.participant_id = assignment_participant.id;
+            ua.lis_result_source_did = lis_result_sourcedid;
+            ua.tenant_id = @tenant.id;
+            ua.save; # Will be used later to post grades back.
+            @tenant.save;
+          end
+          redirect_to "/student_task/view?id=#{assignment_participant.id}" # Redirect to the assignment viewing page for the user
+        else
+          @message = "Assignment number in the request is invalid"
+          @error_code = "LTI_CUSTOM_PARAMETER_INVALID"
+        end
+      end
+    end
+    if(@error_code != 200)
+      @url = params['launch_presentation_return_url'];
+      flash[:notice] = @message
+    end
+  end
+
+  def back_to_lms
+    (redirect_to (params['url']))
+  end
+
+  private
+
+  def define_attributes
+    attributes = {}
+    attributes["role_id"] = Role.student.id
+    attributes["email"] = params['lis_person_contact_email_primary']
+    # The user's name is set as the initial part of the email address.
+    attributes["name"] =  attributes["email"].split("@")[0] if attributes["email"] != nil?
+    attributes["fullname"] = params['lis_person_name_given']
+    attributes["email_on_submission"] = 1
+    attributes["email_on_review"] = 1
+    attributes["email_on_review_of_review"] = 1
+    attributes
+  end
+
+  def create_new_user(attributes)
+    user = User.new(attributes)
```

Comment:
I don't understand this, why this code is in lti_assignment_users_controller.rb?

Reply:
Moreover, the code is **wrong**.  The super_admin should not be the parent of the user, because then the instructor will not ever be able to impersonate this user.  Rather, the parent should be the creator of the assignment.  From this point in the code, we can determine what the assignment is, and then can check the creator of the assignment, and set that creator as the parent of this user.